antsibull/cli/doc_commands/stable.py

```diff
@@ -242,6 +243,22 @@ def get_collection_contents(plugin_content: t.Mapping[str, t.Mapping[str, t.Mapp
     return collection_plugins
 
 
+def get_collection_namespaces(collection_to_something: t.Mapping[str, t.Any]
+                              ) -> t.Dict[str, t.List[str]]:
+    """
+    Return the plugins which are in each collection.
+
+    :arg collection_to_something: A map mapping collection names to something.
+        For example the result of ``get_collection_contents()``.
+    :returns: Mapping from collection namespaces to list of collection names
+    """
+    namespaces = defaultdict(list)
+    for collection_name in collection_to_something:
+        namespace, name = collection_name.split('.', 1)
+        namespaces[namespace].append(name)
+    return namespaces
+
```

Comment:
I think this would be a more generic function (and then a candidate for a function in the library section of the code) if it took an iterable of `namespace.collection` and returned this mapping.  Then the code which calls it can do:

``` python
namespaces = get_collection_namespaces(collection_to_desc.keys())
```

Reply:
I implemented this in d526f300817ce9c8fde08486b9a8a8275e82bb60 and 36f9b6aa597615c51569a2b622b3961b586f0506.